test/common/http/date_provider_impl_test.cc

```diff
@@ -11,11 +11,63 @@
 #include "gtest/gtest.h"
 
 using testing::NiceMock;
+using testing::Return;
 
 namespace Envoy {
 namespace Http {
 
-TEST(DateProviderImplTest, All) {
+class MockTimeSource : public TimeSource {
+public:
+  MOCK_METHOD(SystemTime, systemTime, ());
+  MOCK_METHOD(MonotonicTime, monotonicTime, ());
+};
+
+TEST(SlowDateProviderImplTest, RequestHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
+  EXPECT_CALL(time_source, systemTime()).WillOnce(Return(time_point));
+
+  SlowDateProviderImpl provider(time_source);
+  TestRequestHeaderMapImpl headers;
+  EXPECT_EQ(nullptr, headers.Date());
+
+  provider.setDateHeader(headers);
+  EXPECT_EQ(headers.Date()->value().getStringView(), "Thu, 01 Jan 1970 00:00:02 GMT");
+}
+
+TEST(SlowDateProviderImplTest, ResponseHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
```

Comment:
@rebello95 This line (and the same in the other failing test) is the issue

the conversion from 2million microseconds to seconds is being lost, it seems there is some ambiguity in constructing a duration this way on Windows and the assumed tick representation and period is not the same as Linux (i.e. SystemTime::duration(2000000) is not assumed to be in microseconds)

Reply:
we fixed this locally by explicitly using either `std::chrono::seconds(2)` or `std::chrono::duration<int64_t, std::micro>(2000000)` as the duration to initialize the time point

Reply:
since `SystemTime::duration` gives us no way of setting the period of our tick representation, and the Windows implementation seems to be matching the system practice of a 100ns period

Reply:
@sunjayBhatia updated - thank you so much for helping debug this! Makes sense.

Reply:
We PRed a format check to guard against this sort of thing, feedback appreciated https://github.com/envoyproxy/envoy/pull/12225